src/ussd_pmtct_rapidpro.js

```diff
@@ -1,24 +1,393 @@
+                        );
+                    }
+                },
+                next: "state_dob_month",
+            });
+        });
+
+        self.add("state_dob_month", function(name) {
+            return new ChoiceState(name, {
+                question: $("What month was she born? Please enter the number that matches the answer."),
+                error: $(
+                    "Sorry we don't understand. Please enter the no. next to the mom's answer."
                 ),
-                next: "state_start"
+                choices: [
+                    new Choice("01", $("Jan")),
+                    new Choice("02", $("Feb")),
+                    new Choice("03", $("Mar")),
+                    new Choice("04", $("Apr")),
+                    new Choice("05", $("May")),
+                    new Choice("06", $("Jun")),
+                    new Choice("07", $("Jul")),
+                    new Choice("08", $("Aug")),
+                    new Choice("09", $("Sep")),
+                    new Choice("10", $("Oct")),
+                    new Choice("11", $("Nov")),
+                    new Choice("12", $("Dec")),
+                ],
+                next: "state_dob_day"
+            });
+        });
+
+        self.add("state_dob_day", function(name) {
+            return new FreeText(name, {
+                question: $(
+                    "On what day was she born? Please enter the day as a number, e.g. 12."
+                ),
+                check: function(content) {
+                    var match = content.match(/^(\d+)$/), dob;
+                    if(
+                        !match ||
+                        !(dob = new moment(
+                            self.im.user.answers.state_dob_year + 
+                            self.im.user.answers.state_dob_month + 
+                            match[1], 
+                            "YYYYMMDD")
+                        ) ||
+                        !dob.isValid()
+                    ){
+                        return $(
+                            "Sorry, we don't understand. Please try again by entering the day " +
+                            "she was born as a number, e.g. 12."
+                        );
+                    }
+                },
+                next: "state_trigger_rapidpro_flow"
+            });
+        });
+
+        self.states.add("state_no_registration", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text: "Thank you for using MomConnect. Have a lovely day."
+            });
+        });
+
+        self.states.add("state_no_subscription", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text: $(
+                    "Thank you. The mother will receive messages about " + 
+                    "keeping her baby HIV-negative. Have a lovely day."
```

Comment:
Should this state not be the following content? (It's if they're not registered on MomConnect):
```
Welcome to the Department of Health’s MomConnect. To get msgs about keeping your baby HIV-negative, register to MomConnect by dialing *154*550*2# at the clinic.
```

Reply:
thanks, I missed this error